tests/ChannelMessagesTest.php

```diff
@@ -409,6 +409,31 @@ public function testMessageEncodings() {
     }
 
     /**
+     * Test library interoperability by encoding and decoding messages from messages-encoding.json
+     */
+    public function testMessageEncodingInteroperabilityFixture() {
+        $ably = self::$ably;
+        $fixture = json_decode( file_get_contents( __DIR__ . '/../ably-common/test-resources/messages-encoding.json' ) );
+
+        foreach ($fixture->messages as $i => $testMsgData) {
+            $msg = new Message();
+            $msg->data = $testMsgData->data;
+            $msg->encoding = $testMsgData->encoding;
+            
+            $ably->channel("testInterop_$i")->publish($msg);
+            $ret = $ably->channel("testInterop_$i")->history();
+
+            if ( isset( $testMsgData->expectedHexValue ) ) {
+                $expected = pack( 'H*', $testMsgData->expectedHexValue );
+            } else {
+                $expected = $testMsgData->expectedValue;
+            }
+
+            $this->assertEquals($expected, $ret->items[0]->data);
```

Comment:
Unfortunately this is not doing what is required, see https://github.com/ably/ably-ruby/pull/94 and https://github.com/ably/ably-ios/pull/459 for example.  You need to publish a raw message using a raw HTTP lib (i.e. not through the Ably client lib) and then check that using history it's decoded as expected, and then publish using the Ably client lib, and use history using a RAW http lib to check that the JSON matches what is expected.